WordPress/Classes/ViewRelated/Post/Scheduling/PublishDatePickerViewController.swift

```diff
@@ -0,0 +1,204 @@
+            wpAssertionFailure("initial date value missing")
+        }
+        super.init(rootView: PublishDatePickerView(configuration: configuration))
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        title = Strings.title
+    }
+}
+
+/// - warning: deprecated (kahu-offline-mode)
+extension PublishDatePickerViewController {
+    static func make(viewModel: PublishSettingsViewModel, onDateUpdated: @escaping (Date?) -> Void) -> PublishDatePickerViewController {
+        PublishDatePickerViewController(configuration: .init(
+            date: viewModel.date,
+            isRequired: viewModel.isRequired,
+            timeZone: viewModel.timeZone,
+            updated: onDateUpdated
+        ))
+    }
+}
+
+/// - warning: deprecated (kahu-offline-mode)
+final class PublishDatePickerHelper: NSObject {
+    @objc class func makeDatePicker(post: AbstractPost, isRequired: Bool) -> UIViewController {
+        var viewModel = PublishSettingsViewModel(post: post)
+        return PublishDatePickerViewController.make(viewModel: viewModel) { date in
+            viewModel.setDate(date)
+        }
+    }
+}
+
+struct PublishDatePickerView: View {
+    @State var configuration: PublishDatePickerConfiguration
+
+    var body: some View {
+        Form {
+            Section {
+                dateRow
+                datePickerRow
+                if let date = configuration.date, !configuration.isCurrentTimeZone {
+                    makeTimeZoneMismatchWarningView(date: date)
+                }
+            } header: {
+                Color.clear.frame(height: 0) // Reducing the top inset
+            }
+            if !configuration.isCurrentTimeZone {
+                Section {
+                    timeZoneRow
+                }
+            }
+        }
+        .padding(.top, -8)
+        .environment(\.defaultMinListHeaderHeight, 0)
+        .navigationTitle(Strings.title)
+        .navigationBarTitleDisplayMode(.inline)
+        .tint(Color(uiColor: .brand))
+    }
+
+    private var dateRow: some View {
+        HStack {
+            Image(systemName: "calendar")
+                .foregroundStyle(.secondary)
+            VStack(alignment: .leading) {
+                Text(Strings.date)
+                    .font(.subheadline)
+                Text(selectedValue)
+                    .font(.footnote)
+                    .foregroundStyle(.secondary)
+            }
+            .lineLimit(1)
+
+            Spacer()
+
+            if configuration.date != nil, !configuration.isRequired {
+                Button(role: .destructive, action: {
+                    configuration.date = nil
+                }) {
+                    Image(systemName: "minus.circle.fill")
+                        .foregroundStyle(.red)
+                }
+                .buttonStyle(.plain)
+            }
```

Comment:
Wdyt of changing the icon to "xmark.circle.fill", and not going back to the previous view controller on tap? Imo staying the in same context when you tap the icon makes it more clear that you're resetting the publish date to "Immediately". Here's what I'm envisioning: 

https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/10bc2b55-df3c-444d-8f34-23b4af5b1271

Curious to hear your thoughts, @osullivanchris 🙇‍♀️

Reply:
> Wdyt of changing the icon to "xmark.circle.fill"

Sure, I think it works well. I don't have a strong preference.

> staying the in same context

I added this behavior in one of the latest PRs – it felt a bit too clunky _not_ to pop it, but I'm unsure what's best.

Reply:
@momo-ozawa I switch to a gray "xmark" as you suggested, and it looks pretty good. With an "xmark", staying on the screen now feels like an expected behavior, so I added it to.

Reply:
Thank you!

Reply:
@momo-ozawa I'm a bit out of the loop but that behaviour makes sense to me. I understand clearly what is happening. I think it helps to stay in the same location.

@kean just thought the text showing me the selected scheduled date/time is quite small and the layout is a little bit tight with the icon and 2 pieces of text there. Also the 'Publish date' title is duplicated. I tried a few variations. Only spent 5min on it so its a bit wireframey. But some options if you agree the date could be a bit larger and more prominent. 

This is small fish anyway - not a must fix.

<img width="1344" alt="Screenshot 2024-04-24 at 17 26 57" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/50576199/63d513c5-d3fb-4997-b77d-d4fcc8455b9e">